internal/gcsx/syncer_test.go

```diff
@@ -161,6 +162,63 @@ func (t *FullObjectCreatorTest) CallsCreateObjectsWithObjectProperties() {
 	ExpectEq("test_value", req.Metadata["test_key"])
 }
 
+func (t *FullObjectCreatorTest) CallsCreateObjectWhenSrcObjectIsNil() {
```

Comment:
nit: From a unit testing POV, should we add a test case where we pass non-nill srcObject and a fileName different from srcObject.Name ?

Reply:
I am not sure why should we add a test case for a flow which doesn't exist and check its not getting triggered.